src/Database/DBStructure.php

```diff
@@ -684,6 +684,141 @@ private static function groupBy($indexname, $fieldnames) {
 		return $sql;
 	}
 
+	/**
+	 * 	Check if a table exists
+	 *
+	 * @param string $table Table name
+	 *
+	 * @return boolean Does the table exist?
+	 */
+	public static function existsTable($table)
+	{
+		if (empty($table)) {
+			return false;
+		}
+
+		$table = DBA::escape($table);
+
+		$sql = "SHOW TABLES LIKE '" . $table . "';";
+
+		$stmt = DBA::p($sql);
+
+		if (is_bool($stmt)) {
+			$retval = $stmt;
+		} else {
+			$retval = (DBA::num_rows($stmt) > 0);
+		}
+
+		DBA::close($stmt);
+
+		return $retval;
+	}
+
+	/**
+	 * 	Check if the columns of the table exists
+	 *
+	 * @param string $table   Table name
+	 * @param array  $columns Columns to check ( Syntax: [ $col1, $col2, .. ] )
+	 *
+	 * @return boolean Does the table exist?
+	 */
+	public static function existsColumn($table, $columns = []) {
+		if (empty($table)) {
+			return false;
+		}
+
+		if (is_null($columns) || empty($columns)) {
+			return self::existsTable($table);
+		}
+
+		$table = DBA::escape($table);
+
+		foreach ($columns AS $column) {
```

Comment:
@annando do you know a better way to check if columns exist in a table? I know that a foreach loop for each column isn't the best way..